analyzer/src/main/kotlin/managers/Pub.kt

```diff
@@ -580,15 +581,27 @@ class Pub(
         return yamlMapper.readTree(definitionFile)
     }
 
+    override fun getVersion(workingDir: File?): String {
+        val result = ProcessCapture(workingDir, command(workingDir), getVersionArguments())
+        if (result.isError) {
+            throw IOException(result.errorMessage)
+        }
+
+        return transformVersion(result.stderr)
+    }
+
     override fun command(workingDir: File?): String =
-        if (pubAbsolutePath.isDirectory) "$pubAbsolutePath${File.separator}$pubCommand" else pubCommand
+        if (pubAbsolutePath.isDirectory) "$pubAbsolutePath${File.separator}$dartCommand" else dartCommand
+
+    private fun commandPub(): String =
+        if (pubAbsolutePath.isDirectory) "$pubAbsolutePath${File.separator}$dartCommand pub" else "$dartCommand pub"
 
     private fun commandFlutter(): String =
         if (flutterAbsolutePath.isDirectory) "$flutterAbsolutePath${File.separator}$flutterCommand packages"
         else "$flutterCommand packages"
 
     override fun run(workingDir: File?, vararg args: String): ProcessCapture {
-        var result = ProcessCapture(workingDir, *command(workingDir).split(' ').toTypedArray(), *args)
+        var result = ProcessCapture(workingDir, *commandPub().split(' ').toTypedArray(), *args)
```

Comment:
Can we inline `commandPub()` to just `ProcessCapture(workingDir, command(), "pub", *args)`?

Reply:
It would be possible, but I prefer to explicitly specify a separate method for the pub command, as this is the main package manager command.